tests/framework/data/ActiveDataProviderTest.php

```diff
@@ -177,4 +178,42 @@ public function testPaginationBeforeModels()
         $provider->refresh();
         $this->assertEquals(2, count($provider->getModels()));
     }
+
+    public function testZeroModels()
```

Comment:
What about `testDoesNotPerformQueryWhenHasNoModels`? This seems to be more like a real English sentence.